src/Stats/index.jsx

```diff
@@ -1,33 +1,103 @@
-import React from 'react';
-import { View, ScrollView } from 'react-native';
-import { ListItem } from 'react-native-elements';
+import React, { useState } from 'react';
+import { View, ScrollView, Text } from 'react-native';
+import { ListItem, Overlay, Button } from 'react-native-elements';
 import { useSelector } from 'react-redux';
-import moment from 'moment';
-import utils from './utils';
 import styles from '../Common/themeStyle';
+import MainHeader from '../Common/MainHeader';
+import LargeButton from '../Common/LargeBtn';
+import GradientIcon from '../Common/icon';
+import utils from './utils';
+import SubHeader from '../Common/SubHeader';
+
+const moment = require('moment');
 
 export default function Stats() {
   const { transactions } = useSelector((state) => state.transactions);
+  const [currentMonthSet, setCurrentMonthSet] = useState(utils.getDateSet(moment(), 'months'));
+  const [currentYearSet, setCurrentYearSet] = useState(utils.getDateSet(moment(), 'years'));
+  const [view, setCurrentView] = useState('month');
+  const [overlay, setOverlay] = useState(false);
+
+
+  const ifTransition = (arr) => (arr.length !== 0);
+
+  const updateCurrent = (value, type) => {
+    switch (type) {
+      case 'month':
+        setCurrentMonthSet(utils.getDateSet(value, 'months'));
+        break;
+      case 'year':
+        setCurrentYearSet(utils.getDateSet(value, 'years'));
+        break;
+      default:
+        break;
+    }
+  };
+
+  const updateOverlay = () => {
+    if (overlay === true) {
+      setOverlay(false);
+    } else {
+      setOverlay(true);
+    }
+  };
+
+  const updateHeaderView = (type) => {
+    setCurrentView(type);
+    updateOverlay();
+  };
+
+  const updateListView = (dataList) => {
```

Comment:
it is filtering not updating, and we should highlight in the name what is our filtering based on

Reply:
Done